index.js

```diff
@@ -203,7 +203,9 @@
       return requirements[_name] === Constructor ?
         function(typeRep) {
           return funcPath([name], typeRep) ||
-                 implPath([/function (\w*)/.exec(typeRep)[1], name]);
+                 implPath(typeof typeRep === 'function' ?
+                          [/function (\w*)/.exec(typeRep)[1], name] :
+                          [name]);
```

Comment:
I'm also going to add a test for `Function#name` for environments that support it.